jerry-debugger/src/lib/cdt-controller.ts

```diff
@@ -100,17 +174,34 @@ export class CDTController {
     this.protocolHandler!.stepOut();
   }
 
-  cmdPause() {
-    this.protocolHandler!.pause();
-  }
-
-  cmdResume() {
-    this.protocolHandler!.resume();
+  cmdStepOver() {
+    this.protocolHandler!.stepOver();
   }
 
   // 'report' functions are events from Debugger to CDT
-  private sendPaused(breakpoint: Breakpoint) {
+  private sendPaused(breakpoint: Breakpoint | undefined, backtrace: Array<Breakpoint>) {
     // Node uses 'Break on start' but this is not allowable in crdp.d.ts
-    this.proxyServer!.sendPaused(breakpoint, 'other');
+    const reason = breakpoint ? 'debugCommand' : 'other';
+    this.proxyServer!.sendPaused(breakpoint, backtrace, reason);
+  }
+
+  /**
+   * Enables or disables a breakpoint.
+   *
+   * @param scriptId 1-based script ID
+   * @param line     1-based line number
+   * @param enable   true to enable, false to disable
+   */
+  private updateBreakpoint(scriptId: number, line: number, enable: boolean) {
+    const params: FindBreakpointParams = {
+      scriptId: scriptId,
+      line: line,
+      column: 0,  // TODO: use column
+    };
+    const breakpoint = this.protocolHandler!.findBreakpoint(params);
```

Comment:
Why not call `findBreakpoint` as part of `updateBreakpoint` instead of having the caller to find it?

I'm also a bit puzzled why you're not passing the params as normal arguments instead of wrapping them in an object.

Reply:
To reuse `updateBreakpoint` - it's called elsewhere were the breakpoint is obtained by a different method other than 'findBreakpoint`. The `findBreakpoint` is something to be avoided because it's a slow search.

Reply:
On the params, I'm probably overgeneralizing based on a previous comment you had for me. This might have been where you told me to use an options argument for command line options, to preserve extensibility. But that makes more sense for the case of command line args that could be subject to a lot of change.

Then I probably saw how the Cdrp defines every API in this way, and decided it must be the way "JavaScript people do things". :) But there again I suppose they're trying to make it more likely that your code keeps working when Google adds an optional argument to one of those APIs.

I think I get it better now - for internal APIs of my own that doesn't make sense,  and I'll back out some of that. :)